drivers/usb_c/tcpc/ucpd_stm32.c

```diff
@@ -1022,43 +1081,56 @@ static void ucpd_isr(const struct device *dev_inst[])
 	 * Multiple UCPD ports are available
 	 */
 
-	uint32_t ucpd_base;
+	uint32_t sr0;
+	uint32_t sr1;
+	uint32_t sr0_idx;
+	uint32_t sr1_idx;
 
 	/*
 	 * Since the UCPD peripherals share the same interrupt line, determine
 	 * which one generated the interrupt.
 	 */
-	if (LL_SYSCFG_IsActiveFlag_UCPD1()) {
+
+	/* Map device instance to UCPD device */
+	if ((uint32_t)
+		((const struct tcpc_config *)dev_inst[0]->config)->ucpd_port == UCPD1_BASE) {
+		sr0_idx = 0;
+		sr1_idx = 1;
+	} else {
+		sr0_idx = 1;
+		sr1_idx = 0;
+	}
```

Comment:
This isn't required. You don't need to do any mapping.
The previous code was correct, the only issue was the comment part which was mapping [0] on UCPD1, and the use of `sr1/sr2` which was also misleading.

Reply:
Understood. I removed the change and comment.